test-harness/index.ts

```diff
@@ -1,15 +1,39 @@
-import { parseSpecFile } from './helpers';
+import * as Ajv from 'ajv';
+import { ChildProcess, spawn, spawnSync } from 'child_process';
 import * as fs from 'fs';
-import * as os from 'os'
-import * as path from 'path';
-import * as tmp from 'tmp';
-import { ChildProcess, spawnSync, spawn } from 'child_process'
-import * as split2 from 'split2'
 import { validate } from 'gavel';
 import { parseResponse } from 'http-string-parser';
+import * as os from 'os';
+import * as path from 'path';
+import * as split2 from 'split2';
+import * as tmp from 'tmp';
+import * as toJsonSchema from 'to-json-schema';
+import { parseSpecFile } from './helpers';
 
 jest.setTimeout(60000);
 
+const dynamicHandlers = [
+  {
+    shouldHandle: value => value.includes('.dynamic-response.'),
+    handle: (output, expected) => {
+      const schema = toJsonSchema(JSON.parse(expected.body));
```

Comment:
Hmm, I see two problems here:

1. The generated JSON Schema will never be precise as the original one. If I'm returning

```json
{ "test": "hello" }
```

1. The library will probably generate a `type: "string"` and most likely miss other informations, such as this value been an enum or something like that
2. Since the schema is generated *from the payload* the check will pass, by definition. Therefore this check is kind of useless.